test/module/irohad/ametsuchi/postgres_query_executor_test.cpp

```diff
@@ -977,18 +985,478 @@ namespace iroha {
+          }
+          const rapidjson::Value &json_data_by_writer = json_writer_it->value;
+          if (not json_data_by_writer.IsObject()) {
+            ADD_FAILURE() << "JSON entity for writer " << ref_writer
+                          << " must be an object!";
+            continue;
+          }
+          const auto json_data_by_writer_obj = json_data_by_writer.GetObject();
+
+          EXPECT_EQ(json_data_by_writer_obj.MemberEnd()
+                        - json_data_by_writer_obj.MemberBegin(),
+                    ref_data_by_writer.size())
+              << "Wrong number of keys!";
+
+          // check the values
+          for (const auto &key_and_value : ref_data_by_writer) {
+            const auto &ref_key = key_and_value.first;
+            const auto &ref_val = key_and_value.second;
+
+            const auto it = json_data_by_writer_obj.FindMember(ref_key);
+            if (it == top_obj.MemberEnd()) {
+              ADD_FAILURE() << ref_key << " for writer " << ref_writer
+                            << " not present in JSON!";
+            } else {
+              const rapidjson::Value &data_by_key = it->value;
+              if (not data_by_key.IsString()) {
+                ADD_FAILURE() << "JSON entity for writer " << ref_writer
+                              << ", key " << ref_key << " must be a string!";
+              } else {
+                EXPECT_EQ(data_by_key.GetString(), ref_val);
+              }
+            }
+          }
+        }
+      }
+
+      /**
+       * Query account details and validate the response.
+       */
+      template <typename... Args>
+      auto queryPageAndValidateResponse(Args... args)
+          -> decltype(queryPage(std::declval<Args>()...)) {
+        auto response = queryPage(args...);
+        validatePageResponse(response, args...);
+        return response;
+      }
+    };
+
+    /**
+     * @given account with 9 details from 3 writers, 3 unique keys from each,
+     * and all related permissions
+     * @when queried account details with page metadata not set
+     * @then all 9 detail records are returned and are valid
+     */
+    TEST_F(GetAccountDetailPagedExecutorTest, NoPageMetaData) {
+      addDetails(3, 3);
+
+      shared_model::proto::Query query{[] {
+        iroha::protocol::Query query;
+
+        // set creator account
+        query.mutable_payload()->mutable_meta()->set_creator_account_id(
+            account_id);
+
+        // make a getAccountDetail query
+        query.mutable_payload()->mutable_get_account_detail()->set_account_id(
+            account_id);
+
+        return shared_model::proto::Query{query};
+      }()};
+
+      // send the query
+      QueryExecutorResult response = executeQuery(query);
+
+      // validate result
+      validatePageResponse(
+          response, boost::none, boost::none, boost::none, boost::none, 3 * 3);
+    }
+
+    /**
+     * @given account with 9 details from 3 writers, 3 unique keys from each,
```

Comment:
```suggestion
     * @given account with 1 details from 1 writer, 1 unique key,
```
? because ` addDetails(1, 1);` is called in the test